internal/provider/config_resource.go

```diff
@@ -368,42 +407,60 @@ func (m *MTEConfigResourceModel) transformToApiRequestBody() client.MTEConfigDto
 			Password: m.Config.BasicAuth.Password.ValueString(),
 		}
 	}
+	if m.Config.Cache != nil {
+		var httpCache = make([]client.CacheDto, len(m.Config.Cache))
+
+		for i, r := range m.Config.Cache {
+			var cacheBody = client.CacheDto{
+				TtlSeconds: r.TtlSeconds.ValueInt64Pointer(),
+			}
+			if r.Keys != nil {
+				var cacheKeyHeaders = make([]string, len(r.Keys.Headers))
+				var cacheKeyCookies = make([]string, len(r.Keys.Cookies))
+				for i, h := range r.Keys.Headers {
+					cacheKeyHeaders[i] = h.ValueString()
+				}
+				for i, c := range r.Keys.Cookies {
+					cacheKeyCookies[i] = c.ValueString()
+				}
+				cacheBody.Keys = &client.CacheKeyDto{
+					Header: cacheKeyHeaders,
+					Cookie: cacheKeyCookies,
+				}
+			}
+			if r.PathRules != nil {
+				var anyMatch = make([]string, len(r.PathRules.AnyMatch))
+				var noneMatch = make([]string, len(r.PathRules.NoneMatch))
+				for i, h := range r.PathRules.AnyMatch {
+					anyMatch[i] = h.ValueString()
+				}
+				for i, c := range r.PathRules.NoneMatch {
+					noneMatch[i] = c.ValueString()
+				}
+				cacheBody.PathRules = &client.MatcherDto{
+					AnyMatch:  anyMatch,
+					NoneMatch: noneMatch,
+				}
+			}
+			httpCache[i] = cacheBody
+		}
+		dto.Cache = httpCache
```

Comment:
Maybe this (and its equivalent in the other transformation) should be moved into their own functions, to stop the transformation functions getting too chunky

Reply:
Good shout, updated